go/pkg/rexec/rexec.go

```diff
@@ -130,7 +130,7 @@ func (ec *Context) downloadOutErr() *command.Result {
 func (ec *Context) downloadOutputs(execRoot string) (*rc.MovedBytesMetadata, *command.Result) {
```

Comment:
we override execroot in DownloadOutputs. I'll leave this as is now since we still support execRoot rooted downloads, and we can clean this up when we remove the new option. That way we can check for the option in one central place: downloadOutputs.

Reply:
TL;DR My argument is that this shouldn't be named `execRoot`, just `outDir`, like in the `cas.go` functions.

I think I was confused myself and thus ambiguous - the Download functions in `rexec.go` are context aware, so they do need to know the exec root and the working directory as part of the general context of the action. But the _remote_ location of an artifact does not need to be the _local_ location of an artifact. So this argument shouldn't be _named_ execRoot.

In retrospect, to avoid ambiguity, we probably should use the name `downloadDir` or `downDir` rather than `outDir` everywhere relevant, but I'm ok if you'd rather not go around changing the code once again after I told you to do it once :sweat_smile: